src/App.tsx

```diff
@@ -1,61 +1,152 @@
-import React from 'react';
+import React, { useState, useEffect, useMemo } from 'react';
 import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
+import cn from 'classnames';
 
-import classNames from 'classnames';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 
+import { getUsers } from './api/users';
+import { getPosts } from './api/posts';
+import { User } from './types/User';
+import { Post } from './types/Post';
+
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [selectedUserId, setSelectedUserId] = useState<number | null>(null);
+  const [isPostsLoading, setIsPostsLoading] = useState(false);
+  const [isLoadingPostsFinish, setIsLoadingPostsFinish] = useState(false);
+  const [isErrorOnPostsLoading, setIsErrorOnPostsLoading] = useState(false);
+  const [selectedPostId, setSelectedPostId] = useState<number | null>(null);
+
+  const loadUsers = async () => {
+    try {
+      const loadedUsers = await getUsers();
+
+      setUsers(loadedUsers);
+    } catch (error) {
+      throw new Error('Can\'t load users');
+    }
+  };
+
+  useEffect(() => {
+    loadUsers();
+  }, []);
+
+  const loadPosts = async (userId: number) => {
+    try {
+      setIsErrorOnPostsLoading(false);
+      setIsPostsLoading(true);
+      setIsLoadingPostsFinish(false);
+
+      const loadedPosts = await getPosts(userId);
+
+      setPosts(loadedPosts);
+    } catch (error) {
+      setIsErrorOnPostsLoading(true);
+    } finally {
+      setIsPostsLoading(false);
+      setIsLoadingPostsFinish(true);
+    }
+  };
+
+  useEffect(() => {
+    if (selectedUserId) {
+      loadPosts(selectedUserId);
+    }
+  }, [selectedUserId]);
+
+  const selectedPost = useMemo(() => (
+    posts.find(post => post.id === selectedPostId)
+  ), [selectedPostId]);
+
+  const isNoPostsYet = useMemo(() => (
+    isLoadingPostsFinish && !isErrorOnPostsLoading && posts.length === 0
+  ), [posts, isLoadingPostsFinish, isErrorOnPostsLoading]);
+
+  const isSidebarOpened = useMemo(() => (
+    selectedPost && selectedUserId && selectedPost.userId === selectedUserId
+  ), [selectedPost, selectedUserId]);
+
+  const isPostsVisible = useMemo(() => (
+    posts.length > 0 && isLoadingPostsFinish
+  ), [posts, isLoadingPostsFinish]);
+
```

Comment:
in cases when we use simple calculations using useMemo callback is more "expensive", than the calculation. So when we use filters, or complicated calculations it is the case to wrap it with useMemo, but when it is just a boolean comparison - it is redundant
```suggestion
  const isNoPostsYet = useMemo(() => (
    isLoadingPostsFinish && !isErrorOnPostsLoading && posts.length === 0
  ), [posts, isLoadingPostsFinish, isErrorOnPostsLoading]);

  const isSidebarOpened = useMemo(() => (
    selectedPost && selectedUserId && selectedPost.userId === selectedUserId
  ), [selectedPost, selectedUserId]);

  const isPostsVisible = useMemo(() => (
    posts.length > 0 && isLoadingPostsFinish
  ), [posts, isLoadingPostsFinish]);

```